### PR TITLE
Add iota signature scheme for MAM legacy 

### PR DESCRIPTION
**Pull request by muXxer**
# Description

This PR adds a different ISS (iota signature scheme) which is used in MAM legacy.
The merkle tree for MAM also uses that scheme.

# Type of change

Please delete options that are not relevant.

- [x] Enhancement (non-breaking change which adds functionality)

# How Has This Been Tested?

I added test cases for the new signing scheme.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have written example code according to the contribution guidelines
- [x] I have added tests using ginkgo that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: consts/consts.go [hunk 1]
```diff
@@ -129,3 +129,10 @@ const (
 	Radix                int8 = 3
 	NumberOfTritsInAByte      = 5
 )
+
+// Merkle constants.
+const (
+	IssFragments   uint64 = 27
```
Comment (luca-moser) [inline consts/consts.go]:
```suggestion
	ISSFragments   uint64 = 27
```

File: consts/consts.go [hunk 2]
```diff
@@ -129,3 +129,10 @@ const (
...
+	IssKeyLength   uint64 = HashTrinarySize * IssFragments
```
Comment (luca-moser) [inline consts/consts.go]:
```suggestion
	ISSKeyLength   uint64 = HashTrinarySize * IssFragments
```

File: consts/consts.go [hunk 3]
```diff
@@ -129,3 +129,10 @@ const (
...
+	IssChunkLength int    = HashTrinarySize / TrinaryRadix
```
Comment (luca-moser) [inline consts/consts.go]:
```suggestion
	ISSChunkLength int    = HashTrinarySize / TrinaryRadix
```

File: signing/legacy/signing.go [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+	"errors"
+	"math"
+
+	. "github.com/iotaledger/iota.go/consts"
+	"github.com/iotaledger/iota.go/curl"
+	"github.com/iotaledger/iota.go/kerl"
+	. "github.com/iotaledger/iota.go/trinary"
+)
+
+type SpongeFunctionCreator func() SpongeFunction
```
Comment (luca-moser) [inline signing/legacy/signing.go]: There seems a lot of copy and paste from the original `signing.go` file. Please consider moving the same code into a separate package or use the existing ones.

**Comment by muXxer** [comment]
Good point! Made the changes!

**Comment by luca-moser** [comment]
:+1:

**Pull request merged**